process/block/shardblock.go

```diff
@@ -1835,6 +1835,41 @@ func (sp *shardProcessor) MarshalizedDataToBroadcast(
 	return mrsData, mrsTxs, nil
 }
 
+// DecodeBlockBodyAndHeader method decodes block body and header from a given byte array
+func (sp *shardProcessor) DecodeBlockBodyAndHeader(dta []byte) (data.BodyHandler, data.HeaderHandler) {
```

Comment:
duplicated code with metaBlock.it can be implemented in base. the only difference is to unmarshal into Header or MetaBlock, but both implement HeaderHandler

Reply:
We should unmarshal in concrete type so it should be created another method in both (shardBlock and metaBlock) for the unmarshal execution